File: lab1/q4.java
import java.util.*;
class q4{
	public static void main(String[] args){

		HashMap<Integer,Integer> m = new HashMap<Integer,Integer>();
		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt();
		int arr[] = new int[n];
		for(int i=0;i<n;i++){
			int temp = sc.nextInt();
			arr[i] = temp;
			m.put(temp,i);			
		}
		int k = sc.nextInt();
		boolean flag = false;
		for(int i=0;i<n;i++){
			if(m.containsKey(k-arr[i]) && m.get(k-arr[i])!=i){
				System.out.println("Yes");
				flag = true;
				break;
			}
		}
		if(flag==false)
			System.out.println("No");
	}
}

File: lab1/q4.py
print("Enter size of array")
n = int(input())
print("Enter array elements")
list1 = []
for i in range(0,n):
	temp = int(input())
	list1.append(temp)
print("Enter k")
k = int(input())
flag = False
for i in range(0,n):
	for j in range(i+1,n):
		if(list1[i]+list1[j]==k):
			flag = True;
			break;
if(flag==True):
	print("Yes")
else:
	print("No")
